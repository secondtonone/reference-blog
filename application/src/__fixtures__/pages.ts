import {
  PageViewModel
} from '~/view-model';
import fixtureArticle from '~/__fixtures__/article-content';

const pages: Array<PageViewModel> = [
  {
    id: 1,
    previewImage: '/previewImage.jpg',
    title: 'Why Googleâ€™s 10 Blue Links Will Have to Go (Eventually)',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
      'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog-rc/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/december-2021-product-review-update',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 2,
    previewImage: '/previewImage.jpg',
    title: 'SEO Clinic Episode III: How to Analyze Your Place in the Online Market and Unwrap New Growth Avenues',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/unwrap-new-growth-avenues',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 3,
    previewImage: '/previewImage.jpg',
    title: 'SEO Clinic Episode II: Hands-on Landing Page Optimization Tips',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/hands-on-landing-page-optimization-tips',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 4,
    previewImage: '/previewImage.jpg',
    title: '17 SEO Blogs for Everyone from Beginners to Experts',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/seo-blog',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 5,
    previewImage: '/previewImage.jpg',
    title: '4 Ways to Use Your .htaccess File',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/htaccess-file',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 6,
    previewImage: '/previewImage.jpg',
    title: 'How to Boost Your E-commerce Store\'s SEO by Offering Promo Codes',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/bing-vs-google',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 7,
    previewImage: '/previewImage.jpg',
    title: 'Are Core Updates Becoming Less Powerful? [Study]',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/google-core-update-trends',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 8,
    previewImage: '/previewImage.jpg',
    title: 'How & Why Usability Will Be the Future of SEO',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/seo-usability',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 9,
    previewImage: '/previewImage.jpg',
    title: 'How To Write Explosive Content: The Do\'s and Don\'ts of Marketing',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/seo-usability',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
  {
    id: 10,
    previewImage: '/previewImage.jpg',
    title: 'Content Spam: How to Detect & Combat Negative SEO Tactics',
    author: {
      id: 111,
      name: 'Olga Testova',
      alias: 'olga-testova',
      email: 'example@example.com',
      position: 'Super seo specialist',
      company: 'Yandex',
      bio:
        'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
      photo: 'https://storage.googleapis.com/blog/testing/user/olga.jpg',
      socials: {
        webSite: 'https://example.com',
        twitter: 'https://twitter.com/hello',
        facebook: '100009766004824',
        instagram: 'hello.hello',
        linkedin: 'chto-chto-942423146',
      },
      isPublic: true
    },
    publishedAt: new Date('2020-01-01').getTime(),
    category: {
      id: 1,
      name: 'name',
      slug: 'slug',
      lang: 'lang',
      children: null
    },
    timeToRead: 1,
    url: 'blog/seo-usability',
    template: 'article-base',
    socialImage: 'socialImage',
    preview: 'preview',
    extraImage: 'extra.png',
    altPreviewImage: 'Preview Image',
    altSocialImage: 'Social Image',
    hidePreviewImage: 0,
    editedAt: new Date('2020-02-02').getTime(),
    metaTitle: 'metaTitle',
    metaDescription: 'metaDescription',
    schemaOrg: 'schemaOrg',
    related: [5916, 5915, 5914, 5913],
    translated: null,
    critical: {
      'article-content': {
        id: 1,
        content: fixtureArticle,
        team: 'team',
        galaxy: 'galaxy'
      }
    },
    alternates: [],
    status: 'published',
    tableOfContent: true
  },
];

export default pages;
