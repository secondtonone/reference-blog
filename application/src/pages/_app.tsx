import '~/styles/layout.css';
import '~/styles/fonts.css';
import '~/styles/normalize.css';
import 'lazysizes';

import { AppProps, NextWebVitalsMetric } from 'next/app';
import Head from 'next/head';
import Script from 'next/script';
import getConfig from 'next/config';
import dynamic from 'next/dynamic';

import * as Sentry from '@sentry/node';
import { Integrations } from '@sentry/tracing';
import { ThemeProvider as ThemeContext } from 'next-themes';
import { sendAnalyticEvent } from '~/analytics';

import themeKey from '~/constants/theme-key';
import SubdomainContext from '~/contexts/subdomain-context';
import PopupProvider from '~/providers/popup-provider';
import { Page } from '~/components/atoms';
import isServer from '~/constants/is-server';

const SvgSprite = dynamic(import('~/components/containers/svg-sprite'), { ssr: false });

const {
  publicRuntimeConfig: {
    GTAG_TRACKING_ID,
    sentryConfig
  }
} = getConfig();

if (sentryConfig.ACTIVE) {
  Sentry.init({
    enabled: true,
    dsn: sentryConfig.DSN,
    // @ts-ignore
    integrations: isServer
      ? [new Sentry.Integrations.Http({ tracing: true })]
      : [new Integrations.BrowserTracing()],
    tracesSampleRate: sentryConfig.TRACES_RATE,
    release: sentryConfig.RELEASE,
    environment: sentryConfig.ENV
  });
}

interface Props extends AppProps {
  err: any
}

const App = ({
  Component, pageProps, err
}: Props) => {
  const language = pageProps?.canonicalSubdomain === 'www' ? 'en' : pageProps.canonicalSubdomain || 'en';
  const hideSubscribe = false;
  const isUserDevice = pageProps?.isUser ?? true;

  return (
    <>
      <Script
        id="gtag-analytics"
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];

            window.dataLayer.push({
              'dimension2': 'Unlogged-User', 
              'dimension3': 'not set', 
              'dimension5': '{"_fbp":"fb.1.1615035967638.1412396531"}',
              'dimension12': 'not set'
            });

            (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
            new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
            j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
            'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
            })(window,document,'script','dataLayer','${GTAG_TRACKING_ID}');
          `
        }}
      />
      <SubdomainContext.Provider
        value={{
          language,
          hideSubscribe,
          isUserDevice
        }}
      >
        <ThemeContext
          disableTransitionOnChange
          storageKey={themeKey}
          defaultTheme="light"
        >
          <PopupProvider>
            <Page>
              <Head>
                <meta name="viewport" content="width=device-width, initial-scale=1" />
              </Head>
              <Component
                {...pageProps}
                err={err}
              />
            </Page>
          </PopupProvider>
        </ThemeContext>
      </SubdomainContext.Provider>
      <Script
        src="https://static.example.com/ref-code-script/js/1.3.2/ref.js"
        strategy="lazyOnload"
        onLoad={() => {
          // @ts-ignore
          if ('ReferralCookieHandler' in window && typeof ReferralCookieHandler?.default === 'function') { ReferralCookieHandler.default(); }
        }}
      />
      <SvgSprite />
    </>
  );
};

export function reportWebVitals({
  id, name, value, label
}: NextWebVitalsMetric) {
  sendAnalyticEvent(label === 'web-vital' ? 'web-vitals' : 'next-custom-metric', {
    label: id,
    action: name,
    value: Math.round(name === 'CLS' ? value * 1000 : value),
  });
}

export default App;
