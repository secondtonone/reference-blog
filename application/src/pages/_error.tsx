import NextErrorComponent from 'next/error';
import { NextPageContext } from 'next';
import * as Sentry from '@sentry/node';
import { Errors } from '~/components/organisms';

const errors = {
  404: 'We can\'t find the page you\'re looking for.',
  500: 'Internal Server Error',
  501: 'Not Implemented',
  502: 'Bad Gateway',
  503: 'Service Unavailable',
  504: 'Gateway Timeout'
};

const CustomError = ({
  statusCode, hasGetInitialPropsRun, err,
}) => {
  if (!hasGetInitialPropsRun && err) {
    // eslint-disable-next-line no-console
    console.error(err);

    Sentry.captureException(err);
  }

  const errorMessage = `${errors[statusCode] ?? 'Oops!'}`;

  if (statusCode === 404) {
    return (
      <Errors.PageError
        statusCode={statusCode}
        heading={errorMessage}
      />
    );
  }

  if (statusCode >= 500 && statusCode < 600) {
    return (
      <Errors.PageError
        heading={`${statusCode} ${errorMessage}`}
      />
    );
  }

  return (
    <Errors.Maintenance />
  );
};

CustomError.getInitialProps = async ({ res, err, asPath }) => {
  const errorInitialProps = await NextErrorComponent.getInitialProps({
    res,
    err,
  } as NextPageContext);

  const customErrorProps = (errorInitialProps as any);

  customErrorProps.hasGetInitialPropsRun = true;
  customErrorProps.asPath = asPath;

  if (err) {
    Sentry.captureException(err);
    await Sentry.flush(2000);
  }

  if (res?.statusCode >= 500 && res?.statusCode < 600) {
    Sentry.captureException(err);
    await Sentry.flush(2000);
  }

  Sentry.captureException(
    new Error(`_error.js getInitialProps missing data at path: ${asPath}`)
  );

  return errorInitialProps;
};

export default CustomError;
