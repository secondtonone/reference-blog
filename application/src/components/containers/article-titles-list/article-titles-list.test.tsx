/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';

import pages from '~/__fixtures__/article-titles-list';

import ArticleTitlesList, { IArticleTitlesListProps } from '.';

const articleTitlesListProps: IArticleTitlesListProps = {
  pages,
};

describe('Match snapshots ArticleTitlesList', () => {
  let props: IArticleTitlesListProps;

  beforeEach(() => {
    props = { ...articleTitlesListProps };
  });

  it('default', () => {
    const { asFragment } = render(
      <ArticleTitlesList
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('without pages', () => {
    delete props.pages;
    const { asFragment } = render(
      <ArticleTitlesList
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should show a current title', () => {
    const title = 'How to Improve SEO Rankings in 11 Steps';
    const { asFragment } = render(
      <ArticleTitlesList
        {...props}
        title={title}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
