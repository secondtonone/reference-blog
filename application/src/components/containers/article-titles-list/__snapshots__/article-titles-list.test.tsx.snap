// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ArticleTitlesList default 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 bUchdV article-titles-list__ArticleTitlesListStyled-sc-gtead9-0 cRxOVh"
    data-test="article-titles-list"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0"
    >
      <h3
        class="text-content__TextStyled-sc-1stkrsz-0 frBSzr"
        data-test="article-titles-list-title"
      >
        Start explore with
      </h3>
    </div>
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 bSvQZL article-titles-list__ArticleTitlesListListStyled-sc-gtead9-1 iKdEIW"
      data-test="article-titles-list-list"
    >
      <ul
        class="list__StyledList-sc-8a55c7-0 cHmrMb"
        data-test="list"
      >
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/what-is-seo"
              title="State of Content Ops & Outsourcing Report 2021 — How Businesses Scale"
            >
              State of Content Ops & Outsourcing Report 2021 — How Businesses Scale
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/learn-seo"
              title="4 Ways to Use Your .htaccess File"
            >
              4 Ways to Use Your .htaccess File
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/seo-best-practices"
              title="Content Spam: How to Detect & Combat Negative SEO Tactics"
            >
              Content Spam: How to Detect & Combat Negative SEO Tactics
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/serp-features-guide"
              title="Google Maps Marketing: The Full Guide"
            >
              Google Maps Marketing: The Full Guide
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/google-search-algorithm"
              title="How to Write Explosive Content: The Do's and Don'ts of Marketing"
            >
              How to Write Explosive Content: The Do's and Don'ts of Marketing
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/metasearch-engine"
              title="What Is Content Localization? How to Build a Strategy"
            >
              What Is Content Localization? How to Build a Strategy
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/metrics-in-google-analytics"
              title="Ways to Build Expertise, Authority"
            >
              Ways to Build Expertise, Authority
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/bing-search"
              title="What Is Enterprise SEO and How Is It Different to Traditional SEO?"
            >
              What Is Enterprise SEO and How Is It Different to Traditional SEO?
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Match snapshots ArticleTitlesList should show a current title 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 bUchdV article-titles-list__ArticleTitlesListStyled-sc-gtead9-0 cRxOVh"
    data-test="article-titles-list"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0"
    >
      <h3
        class="text-content__TextStyled-sc-1stkrsz-0 frBSzr"
        data-test="article-titles-list-title"
      >
        How to Improve SEO Rankings in 11 Steps
      </h3>
    </div>
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 bSvQZL article-titles-list__ArticleTitlesListListStyled-sc-gtead9-1 iKdEIW"
      data-test="article-titles-list-list"
    >
      <ul
        class="list__StyledList-sc-8a55c7-0 cHmrMb"
        data-test="list"
      >
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/what-is-seo"
              title="State of Content Ops & Outsourcing Report 2021 — How Businesses Scale"
            >
              State of Content Ops & Outsourcing Report 2021 — How Businesses Scale
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/learn-seo"
              title="4 Ways to Use Your .htaccess File"
            >
              4 Ways to Use Your .htaccess File
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/seo-best-practices"
              title="Content Spam: How to Detect & Combat Negative SEO Tactics"
            >
              Content Spam: How to Detect & Combat Negative SEO Tactics
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/serp-features-guide"
              title="Google Maps Marketing: The Full Guide"
            >
              Google Maps Marketing: The Full Guide
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/google-search-algorithm"
              title="How to Write Explosive Content: The Do's and Don'ts of Marketing"
            >
              How to Write Explosive Content: The Do's and Don'ts of Marketing
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/metasearch-engine"
              title="What Is Content Localization? How to Build a Strategy"
            >
              What Is Content Localization? How to Build a Strategy
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/metrics-in-google-analytics"
              title="Ways to Build Expertise, Authority"
            >
              Ways to Build Expertise, Authority
            </a>
          </div>
        </li>
        <li
          data-test="list-item"
        >
          <span
            data-list="bullet"
          >
            bullet
          </span>
          <div
            data-list="text"
          >
            <a
              href="/blog/bing-search"
              title="What Is Enterprise SEO and How Is It Different to Traditional SEO?"
            >
              What Is Enterprise SEO and How Is It Different to Traditional SEO?
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Match snapshots ArticleTitlesList without pages 1`] = `<DocumentFragment />`;
